subsys/bluetooth/host/hci_core.c

```diff
@@ -3869,6 +3869,13 @@ static void hci_vs_init(void)
 	struct net_buf *rsp;
 	int err;
 
+	if (IS_ENABLED(CONFIG_BT_HCI_VS_EXT_DETECT) &&
+	    (bt_dev.hci_version < BT_HCI_VERSION_5_0 ||
+	     bt_addr_le_cmp(&bt_dev.id_addr, BT_ADDR_LE_ANY))) {
```

Comment:
Shouldn't this be !bt_addr_le_cmp() ?

Reply:
Ah right no, you want to check that it is *not* ANY, i.e. it's a valid public address?

Reply:
I added a clarifying comment